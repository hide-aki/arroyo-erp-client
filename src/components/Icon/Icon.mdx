---
name: Icon
menu: Components
route: /components/Icon
---

import { Props, Playground } from 'docz';
import Icon from './index';

# Icon

This component is meant to render a custom icon.

<Props of={Icon} />

<Playground>
  <Icon bundle='fontello' glyph='circle' />
</Playground>
