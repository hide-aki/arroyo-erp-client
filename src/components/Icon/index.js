import React, {memo} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Icon = memo(function Icon(props) {
  const classesObj = {
    'rubix-icon': true,
    'form-control-feedback': props.feedback || false,
  };
  if (props.bundle) {
    classesObj[props.bundle] = true;
    classesObj[`icon-${props.bundle}-${props.glyph}`] = true;
  } else {
    classesObj[props.glyph] = true;
  }

  const classes = classNames(classesObj, props.className);

  const _props = {
    ...props,
    className: classes.trim(),
  };

  return <span className={_props.className} style={_props.style} />;
});

Icon.propTypes = {
  className: PropTypes.string,
  feedback: PropTypes.bool,
  bundle: PropTypes.string,
  glyph: PropTypes.string.isRequired,
};

export default Icon;
