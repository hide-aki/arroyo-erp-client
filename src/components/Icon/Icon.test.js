// Container.test.js

import React from 'react';
import Enzyme, {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Icon from './';

Enzyme.configure({adapter: new Adapter()});

describe('<DataContent />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<Icon bundle="fontello" glyph="circle" />);
  });

  it('Renders properly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('It renders an Icon', () => {
    expect(wrapper.find(Icon));
  });
  it('It has some props', () => {
    expect(wrapper.find({bundle: 'fontello', glyph: 'circle'}));
  });
});
