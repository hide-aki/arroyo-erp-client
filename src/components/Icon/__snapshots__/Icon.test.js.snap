// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataContent /> Renders properly 1`] = `
<Memo(Icon)
  bundle="fontello"
  glyph="circle"
>
  <span
    className="rubix-icon fontello icon-fontello-circle"
  />
</Memo(Icon)>
`;
